// viviui/dist/components/featureGrid/feature.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const Feature = (_ref) => {
  let {
    title,
    // 傳入title
    icon,
    // 傳入icon
    children // 傳入chakra能接受的樣式到Box裏

  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "icon", "children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, props, !!icon && /*#__PURE__*/_react.default.createElement(_core.Flex, {
    rounded: "full",
    size: 12,
    bg: "blue.500",
    align: "center",
    justify: "center"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    size: 6,
    color: "white",
    as: icon
  })), !!title && /*#__PURE__*/_react.default.createElement(_core.Heading, {
    as: "h2",
    size: "md",
    fontWeight: "semibold",
    mt: "1em",
    mb: "0.5em"
  }, title), children);
};

var _default = Feature;
exports.default = _default;
