// storybook-addon-next-router/dist/storybook-addon-next-router.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var addonActions = require('@storybook/addon-actions');
var addons = require('@storybook/addons');
var React = _interopDefault(require('react'));
var Router = _interopDefault(require('next/router'));
var routerContext = require('next/dist/next-server/lib/router-context');

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

var withNextRouter = /*#__PURE__*/addons.makeDecorator({
  name: 'NextRouter',
  parameterName: 'nextRouter',
  wrapper: function wrapper(getStory, context, settings) {
    Router.router = _extends({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
      push: function push(url, as, options) {
        addonActions.action('nextRouter.push')(url, as, options);
        return Promise.resolve(true);
      },
      replace: function replace(url, as, options) {
        addonActions.action('nextRouter.replace')(url, as, options);
        return Promise.resolve(true);
      },
      reload: function reload() {
        addonActions.action('nextRouter.reload')();
      },
      back: function back() {
        addonActions.action('nextRouter.back')();
      },
      prefetch: function prefetch(url, asPath, options) {
        addonActions.action('nextRouter.prefetch')(url, asPath, options);
        return Promise.resolve();
      },
      beforePopState: function beforePopState(cb) {
        addonActions.action('nextRouter.beforePopState')(cb);
      },
      events: {
        on: function on(type, handler) {
          addonActions.action('nextRouter.events.on')(type, handler);
        },
        off: function off(type, handler) {
          addonActions.action('nextRouter.events.off')(type, handler);
        },
        emit: function emit(type) {
          addonActions.action('nextRouter.events.emit')(type);
        }
      },
      isFallback: false
    }, settings.options, settings.parameters);
    return React.createElement(routerContext.RouterContext.Provider, {
      value: Router.router
    }, getStory(context));
  }
});

exports.withNextRouter = withNextRouter;
