// viviui/dist/components/oldPictureGroup/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _imageItem = _interopRequireDefault(require("./imageItem"));

var _videoItem = _interopRequireDefault(require("./videoItem"));

var _imagesContent = _interopRequireDefault(require("./imagesContent"));

const OldPictureGroup = (_ref) => {
  let {
    photos,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["photos", "children"]);
  const content = [];
  let itemContent = [];
  let items = [];
  const images = photos.slice(0, 5);
  images.forEach((val, index) => {
    if (val.type === 'video') {
      items.push( /*#__PURE__*/_react.default.createElement(_videoItem.default, {
        url: val.url
      }));
    } else {
      items.push( /*#__PURE__*/_react.default.createElement(_imageItem.default, {
        imageURL: val.url
      }));
    }

    if (images.length - 1 === index && (index + 1) % 3 !== 0) {
      itemContent = [];
      itemContent.push(items);
      content.push(itemContent);
    } // 儲存三個


    if ((index + 1) % 3 === 0) {
      itemContent.push(items);
      content.push(itemContent);
      itemContent = items = [];
    }
  });
  return /*#__PURE__*/_react.default.createElement(_core.Box, null, /*#__PURE__*/_react.default.createElement(_core.Box, props, content.map((val, index) => {
    return /*#__PURE__*/_react.default.createElement(_imagesContent.default, {
      key: index,
      className: `images-total-${val[0].length} king`
    }, val);
  }), photos.length > 5 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    justifyContent: "center",
    mt: 2
  }, /*#__PURE__*/_react.default.createElement(_core.Button, {
    variant: "outline",
    variantColor: "green"
  }, `還有${photos.length - images.length}張照片`)), /*#__PURE__*/_react.default.createElement("style", {
    jsx: "true",
    global: "true"
  }, `
          .grid-images {
            display: flex;
          }

          .grid-images.images-total-2 .img-item,
          .grid-images.images-total-3 .img-item{
            position: absolute;
            top: 0px;
            left: 0px;

            width: 100%;
            height: 100%;

            overflow: hidden;

            display: flex;
            justify-content: center;
            align-items: center;
          }

          .grid-images.images-total-2 .imgs:after,
          .grid-images.images-total-3 .imgs:after {
            content: "";
            display: block;
            padding-bottom: 100%;
          }

          .grid-images .imgs {
            width: 100%;
            position: relative;
            display: inline-block;
            background: white;
          }

          .grid-images.images-total-3 .imgs:nth-child(2n) {
            margin: 0 5px;
          }

          .grid-images.images-total-2 .imgs:first-of-type {
            margin-right: 5px;
          }
        `)));
};

var _default = OldPictureGroup;
exports.default = _default;
