// viviui/dist/components/oldPictureGroup/videoItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _videoReact = require("video-react");

// 圖片列表
const VideoItem = (_ref) => {
  let {
    url,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["url", "children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "div",
    className: "imgs"
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    className: "img-item"
  }, /*#__PURE__*/_react.default.createElement(_videoReact.Player, {
    width: "100%",
    height: "100%",
    playsInline: true,
    fluid: false,
    src: url,
    autoPlay: false,
    muted: true,
    mb: "1px"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, null, /*#__PURE__*/_react.default.createElement(_core.Icon, {
    name: "view",
    size: "24px",
    className: "king"
  })), /*#__PURE__*/_react.default.createElement(_videoReact.BigPlayButton, {
    position: "center",
    disabled: true
  }))), /*#__PURE__*/_react.default.createElement("style", null, `
        .video-react-video{
          position: absolute;
          top: 0;
          left: 0;
          width: 100%;
          height: 100%;
        }
        .video-react{
          display: block;
          vertical-align: top;
          box-sizing: border-box;
          color: #fff;
          background-color: #000;
          position: relative;
          font-size: 10px;
          line-height: 1;
          font-family: serif,Times,Times New Roman;
          -webkit-user-select: none;
          -moz-user-select: none;
          -ms-user-select: none;
          user-select: none;
        }
        .video-react-control-bar.video-react-control-bar-auto-hide {
          display: none;
          width: 100%;
          position: absolute;
          bottom: 0;
          left: 0;
          right: 0;
          height: 3em;
          background-color: rgba(43,51,63,.7);
      }
        .video-react-loading-spinner{
          display: none;
          position: absolute;
          top: 50%;
          left: 50%;
          margin: -25px 0 0 -25px;
          opacity: .85;
          text-align: left;
          border: 6px solid rgba(43,51,63,.7);
          box-sizing: border-box;
          background-clip: padding-box;
          width: 50px;
          height: 50px;
          border-radius: 25px;
        }
        `));
};

var _default = VideoItem;
exports.default = _default;
