// viviui/dist/components/csvTable/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _viviui = require("viviui");

var _csvItem = _interopRequireDefault(require("./csvItem"));

const CsvTable = ({
  tableHead,
  csvContent
}) => {
  // 因為有 memo，prop 沒變化就不 rerender
  // props 有變化也是會 render
  // 它內部 state 有變化，還是會 render
  return /*#__PURE__*/React.createElement(_viviui.Box, {
    border: "1px solid #e6e6e6",
    overflow: "auto",
    minH: "300px",
    maxH: "500px"
  }, /*#__PURE__*/React.createElement(_viviui.Table, {
    w: "auto"
  }, /*#__PURE__*/React.createElement(_viviui.TableHead, {
    background: "#f2f2f2"
  }, /*#__PURE__*/React.createElement(_viviui.TableRow, {
    h: "50px"
  }, /*#__PURE__*/React.createElement(_viviui.TableCell, {
    w: "90px"
  }, /*#__PURE__*/React.createElement(_viviui.Box, {
    textAlign: "center",
    w: "90px"
  }, "\u5E8F\u865F")), tableHead.map((head, index) => /*#__PURE__*/React.createElement(_viviui.TableCell, {
    key: index,
    w: `${head.width}px`
  }, /*#__PURE__*/React.createElement(_viviui.Box, {
    textAlign: "center",
    w: `${head.width}px`
  }, head.title))), /*#__PURE__*/React.createElement(_viviui.TableCell, {
    w: "100px"
  }, /*#__PURE__*/React.createElement(_viviui.Box, {
    textAlign: "center",
    w: "90px"
  }, "\u4E0A\u50B3\u72C0\u614B")))), /*#__PURE__*/React.createElement(_viviui.TableBody, {
    fontSize: "14px"
  }, Object.keys(csvContent).map((csvKey, index) => {
    return /*#__PURE__*/React.createElement(_csvItem.default, {
      key: csvKey,
      index: index,
      csvKey: csvKey,
      tableHead: tableHead,
      data: csvContent[csvKey].data,
      state: csvContent[csvKey].state,
      error: csvContent[csvKey].error
    });
  })))) // <Box border='1px solid #e6e6e6' overflow='auto' minH='300px' maxH='500px'>
  //   <Box>
  //     <Box as='table' border='1' w='auto'>
  //       <Box as='thead' background='#f2f2f2' className='csv-table-head'>
  //         <Box as='tr' h='50px'>
  //           <Box as='th' className='csv-tabel-haed-item'>
  //             <Box textAlign='center' w='90px'>序號</Box>
  //           </Box>
  //           {
  //             tableHead.map((head, index) => (
  //               <Box key={index} as='th' className='csv-tabel-haed-item'>
  //                 <Box textAlign='center' w={`${head.width}px`}>{head.title}</Box>
  //               </Box>
  //             ))
  //           }
  //           <Box as='th' className='csv-tabel-haed-item'>
  //             <Box textAlign='center' w='90px'>上傳狀態</Box>
  //           </Box>
  //         </Box>
  //       </Box>
  //       <Box as='tbody' fontSize='14px'>
  // {
  //   Object.keys(csvContent).map((csvKey, index) => {
  //     return (
  //       <CsvItem
  //         key={csvKey}
  //         index={index}
  //         csvKey={csvKey}
  //         tableHead={tableHead}
  //         data={csvContent[csvKey].data}
  //         state={csvContent[csvKey].state}
  //         error={csvContent[csvKey].error}
  //       />
  //     )
  //   })
  // }
  //       </Box>
  //     </Box>
  //   </Box>
  //   <style jsx global>
  //     {`
  //     .csv-table-head .csv-tabel-haed-item:first-child,
  //     .csv-table-body-content .csv-table-body-item:first-child{
  //       border-left: none;
  //     }
  //     .csv-table-head .csv-tabel-haed-item,
  //     .csv-table-body-content .csv-table-body-item{
  //       border-left: 1px #e6e6e6 solid;
  //       border-bottom: 1px #e6e6e6 solid;
  //     }
  //   `}
  //   </style>
  // </Box>
  ;
};

var _default = CsvTable;
exports.default = _default;
