// viviui/dist/components/fileUploadButton/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@chakra-ui/core");

var _fa = require("react-icons/fa");

const FileUploadButton = ({
  getData,
  disabled = false
}) => {
  const toast = (0, _core.useToast)();
  return /*#__PURE__*/React.createElement(_core.Button, {
    variantColor: "teal",
    disabled: disabled,
    position: "relative"
  }, /*#__PURE__*/React.createElement(_core.Box, {
    as: _fa.FaUpload,
    mr: 2
  }), /*#__PURE__*/React.createElement(_core.Box, {
    as: "span"
  }, "\u9078\u64C7 CSV \u6587\u6A94"), /*#__PURE__*/React.createElement(_core.Input, {
    position: "absolute",
    top: "0px",
    bottom: "0px",
    left: "0px",
    right: "0px",
    opacity: "0",
    type: "file",
    name: "resume",
    accept: ".csv",
    cursor: "pointer",
    disabled: disabled,
    onChange: async e => {
      const input = e.target;

      if ('files' in input && input.files.length > 0) {
        let filename = input.files[0].name;
        filename = filename.substr(filename.lastIndexOf('.') + 1); // 檢測導入文件類型

        if (filename.toLowerCase() !== 'csv') {
          toast({
            title: '文件類型錯誤',
            description: '請使用 csv 文檔',
            status: 'error',
            duration: '3000',
            isClosable: true,
            position: 'top'
          }); // 清空 file 的值

          input.value = '';
          return null;
        }

        const file = input.files[0];
        const reader = new FileReader();
        const csvdata = new Promise((resolve, reject) => {
          reader.onload = event => resolve(event.target.result);

          reader.onerror = error => reject(error);

          reader.readAsText(file);
        });
        csvdata.then(content => {
          if (getData) getData(content); // 提交完成後清空 file 的值

          input.value = '';
        });
      }
    }
  }));
};

var _default = FileUploadButton;
exports.default = _default;
