// viviui/dist/components/footer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _container = _interopRequireDefault(require("../container"));

const Footer = (_ref) => {
  let {
    title,
    context,
    sign,
    noDivider = false,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "context", "sign", "noDivider", "children"]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, !noDivider && /*#__PURE__*/_react.default.createElement(_core.Divider, {
    mt: 24,
    mb: 6
  }), /*#__PURE__*/_react.default.createElement(_container.default, (0, _extends2.default)({
    mb: {
      base: 4,
      md: 20
    },
    px: 4
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "footer"
  }, children)));
};

var _default = Footer;
exports.default = _default;
