// viviui/dist/components/hero/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _container = _interopRequireDefault(require("../container"));

const Hero = (_ref) => {
  let {
    title,
    // 傳入標題
    subtitle,
    // 傳入小標題
    children,
    // children的子孫
    noBottomBorder = false,
    // 是否需要底綫
    leftButton = null,
    // 輸入左邊按鈕
    rightButton = null // 傳入chakra能接受的樣式到Box裏

  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "subtitle", "children", "noBottomBorder", "leftButton", "rightButton"]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "section",
    pt: 40,
    pb: 24
  }, props), /*#__PURE__*/_react.default.createElement(_container.default, null, /*#__PURE__*/_react.default.createElement(_core.Box, {
    maxW: "xl",
    mx: "auto",
    textAlign: "center"
  }, !!title && /*#__PURE__*/_react.default.createElement(_core.Heading, {
    size: "xl",
    fontWeight: "semibold"
  }, title), !!subtitle && /*#__PURE__*/_react.default.createElement(_core.Text, {
    as: "div",
    opacity: "0.7",
    fontSize: "xl",
    mt: title ? 6 : 0
  }, subtitle), (!!leftButton || !!rightButton) && /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: title || subtitle ? 4 : 0
  }, leftButton, rightButton), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: title || subtitle || rightButton || leftButton ? 4 : 0
  }, children)))), !noBottomBorder && /*#__PURE__*/_react.default.createElement(_core.Divider, {
    my: {
      base: 10,
      sm: 10,
      md: 16
    }
  }));
};

var _default = Hero;
exports.default = _default;
