// viviui/dist/components/modalSelector/modalSelectorOption.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

var _modalContext = require("./modalContext");

const ModalSelectorOption = props => {
  // 從 Context 組件取出裏面的值
  const {
    value: selectValue,
    onChange,
    onClose
  } = (0, _react.useContext)(_modalContext.ModalContext);
  return /*#__PURE__*/_react.default.createElement(_core.Button, (0, _extends2.default)({
    as: "div",
    mb: 4,
    variant: "ghost",
    w: "100%",
    justifyContent: "flex-start",
    size: "lg",
    overflow: "hidden",
    whiteSpace: "nowrap",
    textOverflow: "ellipsis",
    border: {
      base: 'none'
    },
    textAlign: "left",
    cursor: "pointer",
    lineHeight: "40px",
    isDisabled: selectValue === props.value,
    onClick: () => {
      // 改變哦那 Change 裏的值
      onChange(props.value); // 判斷從 Context 組件取出的值與點擊 Button 后的值是否一樣，不一樣就執行從 Context 拿出的方法去關閉 modal

      if (selectValue !== props.value) {
        onClose();
      }
    }
  }, props));
};

var _default = ModalSelectorOption;
exports.default = _default;
