// viviui/dist/components/photoMagnify/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

const photoMagnify = (_ref) => {
  let {
    isOpen,
    onClose,
    images = [],
    indexI = 0,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["isOpen", "onClose", "images", "indexI", "children"]);
  const [index, setIndex] = (0, _react.useState)(indexI); // 左右边界

  if (index < 0) {
    setIndex(0);
  }

  if (index > images.length - 1) {
    setIndex(images.length - 1);
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Modal, (0, _extends2.default)({
    isOpen: isOpen,
    onClose: onClose,
    closeOnOverlayClick: false,
    size: "full"
  }, props), /*#__PURE__*/_react.default.createElement(_core.ModalContent, {
    h: "100%",
    my: 0,
    background: "rgba(255, 255, 255, 0.9)"
  }, /*#__PURE__*/_react.default.createElement(_core.ModalCloseButton, {
    size: "lg"
  }), /*#__PURE__*/_react.default.createElement(_core.Flex, {
    justifyContent: "center",
    alignItems: "center",
    h: "50px"
  }, /*#__PURE__*/_react.default.createElement(_core.Text, null, `照片 (${index + 1}/${images.length})`)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    p: 4,
    d: "flex",
    h: "100%",
    w: "100%"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    h: "100%",
    d: "flex",
    alignItems: "center",
    justifyContent: "center"
  }, /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    variantColor: "teal",
    icon: "chevron-left",
    borderRadius: 9999,
    onClick: () => {
      setIndex(index === 0 ? images.length - 1 : index - 1);
    },
    visibility: index === 0 ? 'hidden' : 'visible'
  })), /*#__PURE__*/_react.default.createElement(_core.Box, {
    px: 4,
    h: "100%",
    d: "flex",
    alignItems: "center",
    justifyContent: "center",
    flex: "1"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    h: "100%",
    d: "flex",
    alignItems: "center",
    justifyContent: "center",
    flex: "1"
  }, /*#__PURE__*/_react.default.createElement(_core.Image, {
    src: images[index],
    maxW: "100%",
    maxH: "100%",
    objectFit: "contain",
    ignoreFallback: true
  }))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    h: "100%",
    d: "flex",
    alignItems: "center",
    justifyContent: "center"
  }, /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    variantColor: "teal",
    icon: "chevron-right",
    borderRadius: 9999,
    onClick: () => {
      setIndex(index === images.length - 1 ? 0 : index + 1);
      console.log(flagRef.current);
    },
    visibility: index + 1 === images.length ? 'hidden' : 'visible'
  }))))));
};

var _default = photoMagnify;
exports.default = _default;
