// viviui/dist/components/Fieldset/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

var _leftFieldset = _interopRequireDefault(require("./leftFieldset"));

var _rightFieldset = _interopRequireDefault(require("./rightFieldset"));

var _fieldsetHeader = _interopRequireDefault(require("./fieldsetHeader"));

var _container = _interopRequireDefault(require("../container"));

// 學校專業的 title 底綫模塊
const Fieldset = (_ref) => {
  let {
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);
  const {
    colorMode
  } = (0, _core.useColorMode)();
  let left = null;
  let right = null;
  let title = null;
  const chil = [];
  console.log(children);

  _react.Children.map(children, (c, i) => {
    switch (c.type) {
      case _leftFieldset.default:
        left = c; // 自定义左边数据

        break;

      case _rightFieldset.default:
        right = c; // 自定义右边数据

        break;

      case _fieldsetHeader.default:
        title = c; // 自定义右边数据

        break;

      default:
        chil.push(c);
    }
  });

  return /*#__PURE__*/_react.default.createElement(_container.default, (0, _extends2.default)({
    pb: 4,
    maxW: "960px"
  }, props, {
    bg: colorMode === 'light' ? props.bg || props.background : props.dark || '#1A202C'
  }), /*#__PURE__*/_react.default.createElement(_core.Box, {
    position: "relative",
    width: "full",
    textAlign: "center",
    my: "20px",
    bg: colorMode === 'light' ? props.bg || props.background : props.dark || '#1A202C' // {...props}

  }, left, /*#__PURE__*/_react.default.createElement(_core.Flex, {
    justify: "center",
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Box, {
    borderBottom: `1px solid ${colorMode === 'light' ? '#aaa6a6' : 'white'}`,
    w: "100%",
    position: "absolute",
    zIndex: "0"
  }), title, chil)), right));
};

var _default = Fieldset;
exports.default = _default;
