// viviui/dist/components/postCard/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _postPage = require("./postPage");

var _PostUpperRight = _interopRequireDefault(require("./PostUpperRight"));

var _postTalent = _interopRequireDefault(require("./postTalent"));

var _postContent = _interopRequireDefault(require("./postContent"));

var _photosGrid = _interopRequireDefault(require("./photosGrid"));

const PostCard = (_ref) => {
  let {
    title,
    subtitle,
    postHeaderLeft,
    postUpperRight,
    postMenu,
    tags,
    talentName,
    content,
    creator,
    postPhotos,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "subtitle", "postHeaderLeft", "postUpperRight", "postMenu", "tags", "talentName", "content", "creator", "postPhotos", "children"]);
  const {
    colorMode
  } = (0, _core.useColorMode)();
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_postPage.PostPage, (0, _extends2.default)({
    bg: colorMode === 'light' ? 'white' : '#1A202C'
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "header",
    position: "relative"
  }, /*#__PURE__*/_react.default.createElement(_PostUpperRight.default, null, postMenu), creator, !!talentName && /*#__PURE__*/_react.default.createElement(_postTalent.default, null, talentName), tags, /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: 3
  }, /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontWeight: "bold",
    textAlign: "left",
    minH: "50px",
    wordBreak: "break-word",
    fontSize: "3xl"
  }, title || '無標題'))), children, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article"
  }, !!content && /*#__PURE__*/_react.default.createElement(_postContent.default, null, content), /*#__PURE__*/_react.default.createElement(_photosGrid.default, null, postPhotos))));
};

var _default = PostCard;
exports.default = _default;
