// viviui/dist/components/postCard/postCreatorHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const PostCreatorHeader = (_ref) => {
  let {
    profilePhoto,
    creatorName,
    creatorTime
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["profilePhoto", "creatorName", "creatorTime"]);
  return /*#__PURE__*/_react.default.createElement(_core.Stack, (0, _extends2.default)({
    isInline: true
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "div",
    w: "50px",
    h: "50px",
    rounded: "50%",
    d: "flex",
    alignItems: "center",
    justifyContent: "center",
    overflow: "hidden"
  }, profilePhoto), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "div",
    ml: 2,
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "left"
  }, creatorName, !!creatorTime && /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "md",
    color: "gray.400"
  }, creatorTime)));
};

var _default = PostCreatorHeader;
exports.default = _default;
