// viviui/dist/components/postCard/postTalent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const PostTalent = (_ref) => {
  let {
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: 3
  }, /*#__PURE__*/_react.default.createElement(_core.Box, null, /*#__PURE__*/_react.default.createElement(_core.Tag, {
    rounded: "full",
    variant: "outline",
    variantColor: "green",
    style: {
      whiteSpace: 'nowrap'
    }
  }, /*#__PURE__*/_react.default.createElement(_core.Icon, {
    mr: "1",
    name: "star"
  }), /*#__PURE__*/_react.default.createElement(_core.TagLabel, null, children))));
};

var _default = PostTalent;
exports.default = _default;
