// viviui/dist/components/postCard/postTags.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _fa = require("react-icons/fa");

const PostTags = (_ref) => {
  let {
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    mt: 3,
    d: "flex",
    alignItems: "center",
    flexWrap: "wrap"
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: _fa.FaUserTag,
    fontSize: "20px",
    w: "20px",
    mr: 2
  }), children);
};

var _default = PostTags;
exports.default = _default;
