// viviui/dist/components/table/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

const Table = (_ref) => {
  let {
    type,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["type", "children"]);
  let table = null;
  let stripe = null;
  let line = null;
  let hover = null;
  let condensed = null;

  if (type) {
    type.split(' ').map(c => {
      switch (c) {
        case 'table':
          table = c;
          break;

        case 'stripe':
          stripe = c;
          break;

        case 'condensed':
          condensed = c;
          break;

        case 'hover':
          hover = c;
          break;

        case 'line':
          line = c;
          break;
      }
    });
  }

  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "table",
    className: "table",
    w: "100%",
    letterSpacing: "0.05rem"
  }, props), children, table && /*#__PURE__*/_react.default.createElement("style", null, `
              .table tr td,th{
                border: 1px solid #d0cbcb;
                text-align: center;
                padding: 8px;
              }
              `), hover && /*#__PURE__*/_react.default.createElement("style", null, `
              .table tbody tr:hover{
                  cursor:auto;
                  background-color:rgba(0, 0, 0, 0.05);
              }
              `), stripe && /*#__PURE__*/_react.default.createElement("style", null, `
              .table tbody tr:nth-child(odd) {
                background-color: rgba(0, 0, 0, 0.01);
             }
              `), line && /*#__PURE__*/_react.default.createElement("style", null, `
              .table tr td {
                border-top: 1px solid #f1f2f4;
                text-align: left;
                padding: 8px;
             }
              `), condensed && /*#__PURE__*/_react.default.createElement("style", null, `
              .table tr td,th {
                padding: 5px;
             }
              `));
};

var _default = /*#__PURE__*/(0, _react.memo)(Table);

exports.default = _default;
