// viviui/dist/components/header/headerButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

const HeaderButton = /*#__PURE__*/(0, _react.forwardRef)((_ref, ref) => {
  let {
    title,
    icon,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "icon", "children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Button, (0, _extends2.default)({
    as: "div",
    align: "center",
    fontSize: {
      base: 'xs',
      sm: 'md',
      md: 'lg'
    },
    mx: {
      base: 1,
      sm: 2
    },
    display: {
      base: 'none',
      sm: 'none',
      md: 'flex'
    },
    variant: "ghost",
    color: "gray.500",
    forwardedRef: ref
  }, props), !!icon && /*#__PURE__*/_react.default.createElement(_core.Box, {
    m: 0,
    ml: {
      base: 0,
      md: 1
    },
    mr: {
      base: 0,
      md: 1
    }
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: icon,
    size: "1.5em"
  })), title, children);
});

var _default = /*#__PURE__*/(0, _react.memo)(HeaderButton);

exports.default = _default;
