// viviui/dist/components/avatarTooltip/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _container = _interopRequireDefault(require("../container"));

const AvatarTooltip = (_ref) => {
  let {
    left,
    right,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["left", "right", "children"]);
  return /*#__PURE__*/_react.default.createElement(_container.default, props, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "header",
    textAlign: "center"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "section",
    maxW: "100%",
    mt: "20px",
    d: "flex",
    justifyContent: "center"
  }, left, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "div"
  }, right))));
};

var _default = AvatarTooltip;
exports.default = _default;
