// viviui/dist/components/postCardPicture/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

const postCardPicture = (_ref) => {
  let {
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);
  let a = null;
  let b = null;
  let c = null;
  let d = null;
  let e = null;

  _react.Children.map(children, (chil, i) => {
    switch (i) {
      case 0:
        a = chil; // 自定義logo

        break;

      case 1:
        b = chil; // Header左邊列表

        break;

      case 2:
        c = chil; // Header 中間列表

        break;

      case 3:
        d = chil; // Header 右邊列表

        break;

      case 4:
        e = chil; // Header 右邊列表

        break;
    }
  });

  return /*#__PURE__*/_react.default.createElement(_core.Box, props, /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: "10px",
    mx: {
      base: 'auto',
      sm: '-13px'
    }
  }, children.length < 2 && /*#__PURE__*/_react.default.createElement(_core.Box, null, a), children.length === 2 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, a)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, b))), children.length === 3 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    maxW: "100%",
    maxH: "50%"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    width: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, a)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, b)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 0 0 1px",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, c))), children.length === 4 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    flexWrap: "wrap"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    maxW: "100%",
    maxH: "50%"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    width: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, a)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, b)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 0 0 1px",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, c))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: "5px"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: "100%",
    h: "50%",
    m: "0 0 0 1px",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, d)))), children.length === 5 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    flexWrap: "wrap"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    maxW: "100%",
    maxH: "50%"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    width: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, a)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, b)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 0 0 1px",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, c))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: "5px"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, d)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, e))))), children.length >= 6 && /*#__PURE__*/_react.default.createElement(_core.Box, {
    flexWrap: "wrap"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    maxW: "100%",
    maxH: "50%"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    width: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, a)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 1px 0 0",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, b)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '49.8%', '33.8%'],
    h: "50%",
    m: "0 0 0 1px",
    border: "1px solid #b7b7b7"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    h: "50%",
    ratio: 1
  }, c))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: "5px"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, d)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "article",
    w: ['50%', '50%', '50%', '50%'],
    m: "0 0 0 1px"
  }, /*#__PURE__*/_react.default.createElement(_core.AspectRatioBox, {
    ratio: 1,
    m: "1px 0 0 0",
    border: "1px solid #b7b7b7"
  }, e)))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    justifyContent: "center",
    mt: 2
  }, /*#__PURE__*/_react.default.createElement(_core.Button, {
    variant: "outline",
    variantColor: "green"
  }, `還有${children.length - 5}張照片`)))));
};

var _default = postCardPicture;
exports.default = _default;
