// viviui/dist/components/newModal/modalContent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const ModalWithContent = (_ref) => {
  let {
    mdalHeader,
    closeButton,
    modalWithBody,
    isOpen,
    onClose,
    modalWithFooter,
    children,
    modalbodyProps
  } = _ref,
      rest = (0, _objectWithoutProperties2.default)(_ref, ["mdalHeader", "closeButton", "modalWithBody", "isOpen", "onClose", "modalWithFooter", "children", "modalbodyProps"]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.ModalContent, (0, _extends2.default)({
    h: {
      base: '100%',
      sm: 'auto'
    },
    maxH: "100%",
    my: {
      base: 0,
      sm: '3.75rem'
    }
  }, rest), mdalHeader, !!closeButton && /*#__PURE__*/_react.default.createElement(_core.ModalCloseButton, null), /*#__PURE__*/_react.default.createElement(_core.ModalBody, (0, _extends2.default)({
    pb: 6
  }, modalbodyProps), modalWithBody, children), !!modalWithFooter && /*#__PURE__*/_react.default.createElement(_core.ModalFooter, null, modalWithFooter)));
};

var _default = ModalWithContent;
exports.default = _default;
