// viviui/dist/components/cardGroup/card.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 學校專業的白色小塊 component
const Card = (_ref) => {
  let {
    onClick,
    title,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["onClick", "title", "children"]);
  const {
    colorMode
  } = (0, _core.useColorMode)();
  return /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "a",
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "article",
    w: ['45%', '30.5%', '22.5%', '22.6%'],
    p: "20px",
    borderRadius: "10px",
    d: "inline-block",
    m: ['2%', '1%', '1%', '1%'],
    shadow: colorMode === 'light' ? '2px 2px 6px 0px #b9b9b9' : '2px 2px 6px 0px #2D3748',
    bg: colorMode === 'light' ? props.bg || props.background || 'white' : props.dark || '#1A202C'
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "h2",
    mb: "30%",
    textAlign: "left",
    fontSize: {
      base: '16px',
      sm: '18px',
      md: '20px'
    },
    fontWeight: "bold"
  }, title, children)));
};

var _default = Card;
exports.default = _default;
