// viviui/dist/components/avatarTooltipGroup/avatars.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

var _avatarTooltipContext = require("./avatarTooltipContext");

const Tooltips = (_ref) => {
  let {
    profilePhotoSrc
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["profilePhotoSrc"]);
  const {
    name
  } = (0, _react.useContext)(_avatarTooltipContext.AvatarTooltipContext);
  return /*#__PURE__*/_react.default.createElement(_core.Avatar, (0, _extends2.default)({
    mr: 1,
    w: {
      base: '30px',
      md: '35px'
    },
    h: {
      base: '30px',
      md: '35px'
    },
    name: name || '',
    src: profilePhotoSrc,
    fontSize: {
      base: '13px',
      md: '15px'
    },
    color: "black"
  }, props));
};

var _default = Tooltips;
exports.default = _default;
