// viviui/dist/components/page/deleteRemind.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 刪除提示框
const HeaderPage = (_ref) => {
  let {
    title,
    content,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "content", "children"]);
  const {
    colorMode
  } = (0, _core.useColorMode)();
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    border: "1px solid #FED7D7",
    background: colorMode === 'light' ? '#FFF5F5' : '#2D3748',
    rounded: 4,
    mt: 20,
    p: 5
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, null, /*#__PURE__*/_react.default.createElement(_core.Heading, {
    as: "h4",
    size: "md"
  }, title), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mt: 4
  }, content)), /*#__PURE__*/_react.default.createElement(_core.Divider, null), /*#__PURE__*/_react.default.createElement(_core.Box, {
    textAlign: "right",
    pr: 3,
    pt: 8
  }, children));
};

var _default = HeaderPage;
exports.default = _default;
