// viviui/dist/components/page/manageMent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 學校管理員的數據
const Management = (_ref) => {
  let {
    subTitle,
    children,
    descriptionProps
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["subTitle", "children", "descriptionProps"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "article"
  }, props), !!subTitle && /*#__PURE__*/_react.default.createElement(_core.Heading, (0, _extends2.default)({
    as: "h3",
    fontSize: {
      base: '15px',
      sm: '1rem'
    },
    mt: 10,
    mb: 4
  }, descriptionProps), subTitle), children);
};

var _default = Management;
exports.default = _default;
