// viviui/dist/components/page/row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 顯示學校用戶的component
const Row = (_ref) => {
  let {
    right,
    left,
    name,
    profilePhoto,
    noDivider = false,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["right", "left", "name", "profilePhoto", "noDivider", "children"]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Flex, (0, _extends2.default)({
    justify: "space-between",
    width: "full",
    maxWidth: "1280px",
    mx: "auto",
    px: {
      base: '1.25rem',
      sm: '1.25rem',
      md: '1.25rem',
      lg: 0
    }
  }, props), /*#__PURE__*/_react.default.createElement(_core.Flex, null, /*#__PURE__*/_react.default.createElement(_core.Box, {
    mr: 5
  }, /*#__PURE__*/_react.default.createElement(_core.Avatar, {
    name: name,
    src: profilePhoto
  })), children, /*#__PURE__*/_react.default.createElement(_core.Box, null, left)), /*#__PURE__*/_react.default.createElement(_core.Flex, null, right)), !noDivider && /*#__PURE__*/_react.default.createElement(_core.Divider, {
    my: 6
  }));
};

var _default = Row;
exports.default = _default;
