// viviui/dist/components/page/wbgPage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 白板的component，有backurl時，又有children，就會出現白板，沒有children就不會出現白板
const WbgPage = (_ref) => {
  let {
    title,
    nodivider,
    headerPage,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["title", "nodivider", "headerPage", "children"]);
  const {
    colorMode
  } = (0, _core.useColorMode)();
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    as: "main"
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "div",
    w: {
      base: '100%',
      md: '90%'
    },
    mt: "2px",
    mx: "auto",
    bg: colorMode === 'light' ? props.bg || props.background || 'white' : props.dark || '#1A202C',
    minH: "100vh",
    maxWidth: "928px",
    borderRadius: "4px 4px 0 0",
    position: "relative",
    boxShadow: {
      base: '',
      sm: '',
      md: colorMode === 'light' ? '0 -1px 10px rgba(0,0,0,0.05), 0 1px 4px rgba(0,0,0,0.1), 0 10px 30px #f3ece8' : '0 -1px 10px rgba(0,0,0,0.05), 0 1px 4px rgba(0,0,0,0.1), 0 10px 30px #2D3748'
    }
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    pt: "1px"
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "section",
    mx: "auto",
    w: "100%",
    maxW: {
      base: '100%',
      sm: '100%',
      md: '800px'
    },
    bg: colorMode === 'light' ? props.bg || props.background || 'white' : props.dark || '#1A202C',
    rounded: 8,
    transition: "0.3s"
  }, (!!title || !!headerPage) && /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Box, {
    as: "header",
    pt: 4,
    px: 16,
    alignItems: "center"
  }, title, headerPage), !nodivider && /*#__PURE__*/_react.default.createElement(_core.Divider, {
    my: 6,
    d: {
      base: 'none',
      md: 'flex'
    }
  })), !!children && children))));
};

var _default = WbgPage;
exports.default = _default;
