// viviui/dist/components/fullScreenView/blueButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const BlueButton = (_ref) => {
  let {
    leftIcon,
    rightIcon,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["leftIcon", "rightIcon", "children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Button, (0, _extends2.default)({
    as: "a",
    size: "lg",
    w: {
      base: '98%',
      sm: '340px'
    },
    variantColor: "blue",
    borderRadius: "9999px"
  }, props), leftIcon, /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    flex: "1 0 auto",
    justifyContent: "center",
    mt: "-1px",
    ml: "8px"
  }, children), rightIcon);
};

var _default = BlueButton;
exports.default = _default;
