// viviui/dist/components/fullScreenView/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

var _fullscreenViewContainer = _interopRequireDefault(require("./fullscreenViewContainer"));

var _closeLink = _interopRequireDefault(require("./closeLink"));

const FullscreenView = (_ref) => {
  let {
    closePath,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["closePath", "children"]);
  return /*#__PURE__*/_react.default.createElement(_fullscreenViewContainer.default, props, !!closePath && /*#__PURE__*/_react.default.createElement(_closeLink.default, {
    href: closePath
  }), /*#__PURE__*/_react.default.createElement(_core.Box, {
    w: "100%",
    maxW: "1088px",
    d: "flex",
    alignItems: "center",
    flexDirection: "column",
    py: "32px",
    px: "16px",
    flex: "1 0 auto",
    justifyContent: "center"
  }, children));
};

var _default = FullscreenView;
exports.default = _default;
