// viviui/dist/components/editShow/editTable.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

// 編輯 EditTable
const EditTable = (_ref) => {
  let {
    value,
    inputonChange
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["value", "inputonChange"]);
  return /*#__PURE__*/_react.default.createElement(_core.Editable, (0, _extends2.default)({
    fontSize: {
      base: '20px',
      sm: '20px',
      md: '24px'
    },
    defaultValue: value,
    d: "flex",
    isPreviewFocusable: false,
    onChange: inputonChange
  }, props), ({
    isEditing,
    onSubmit,
    onRequestEdit
  }) => /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, !isEditing && /*#__PURE__*/_react.default.createElement(_core.Flex, {
    align: "center"
  }, /*#__PURE__*/_react.default.createElement(_core.IconButton, {
    variant: "outline",
    size: "xs",
    icon: "edit",
    onClick: onRequestEdit
  })), /*#__PURE__*/_react.default.createElement(_core.Box, {
    p: "5px 10px"
  }, /*#__PURE__*/_react.default.createElement(_core.EditablePreview, {
    maxW: {
      base: '250px',
      sm: '230px',
      md: '340px'
    },
    alignItems: "center",
    whiteSpace: "nowrap",
    d: "block",
    overflow: "hidden",
    textOverflow: "ellipsis",
    justifyContent: "space-between"
  }), /*#__PURE__*/_react.default.createElement(_core.EditableInput, null))));
};

var _default = EditTable;
exports.default = _default;
