// viviui/dist/components/editShow/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@chakra-ui/core");

var _editTable = _interopRequireDefault(require("./editTable"));

var _deleteIcon = _interopRequireDefault(require("./deleteIcon"));

var _cancelButton = _interopRequireDefault(require("./cancelButton"));

var _confirmButton = _interopRequireDefault(require("./confirmButton"));

const EditIntelligence = (_ref) => {
  let {
    left,
    onChage,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["left", "onChage", "children"]);
  const editTable = [];
  let deleteIco = null;
  let cancel = null;
  let confirm = null;
  const right = []; // 把 children 里的标签拿出来后赋值。

  _react.Children.map(children, (c, i) => {
    switch (c.type) {
      case _editTable.default:
        editTable.push(c);
        break;

      case _deleteIcon.default:
        deleteIco = c;
        break;

      case _cancelButton.default:
        cancel = c;
        break;

      case _confirmButton.default:
        confirm = c;
        break;

      default:
        right.push(c);
    }
  });

  return /*#__PURE__*/_react.default.createElement(_core.PseudoBox, (0, _extends2.default)({
    d: {
      base: 'block',
      sm: 'flex'
    },
    p: 2,
    alignItems: "center",
    justifyContent: "space-between",
    _hover: {
      bg: 'blue.100',
      color: 'black'
    }
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    d: "flex",
    justifyContent: "space-between",
    width: {
      base: '90%',
      sm: '64%'
    },
    mx: {
      base: 'auto',
      sm: 'none'
    }
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    ml: {
      base: 0,
      sm: 3
    }
  }, editTable), /*#__PURE__*/_react.default.createElement(_core.Box, {
    my: "auto"
  }, deleteIco)), /*#__PURE__*/_react.default.createElement(_core.Box, {
    width: {
      base: '90%',
      sm: '30%'
    },
    mx: {
      base: 'auto',
      sm: 'none'
    },
    display: "flex",
    alignItems: "center",
    justifyContent: {
      base: 'space-between',
      sm: 'space-around'
    }
  }, cancel, confirm, right));
};

var _default = EditIntelligence;
exports.default = _default;
