// viviui/dist/components/singleAnnouncement/announcementHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const SingleAnnouncement = (_ref) => {
  let {
    content,
    title,
    left,
    right,
    time,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["content", "title", "left", "right", "time", "children"]);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_core.Box, {
    borderBottom: "1px solid #4A5568"
  }, /*#__PURE__*/_react.default.createElement(_core.Heading, {
    as: "h1",
    fontSize: "26px",
    textAlign: "center",
    fontWeight: "bold"
  }, title), /*#__PURE__*/_react.default.createElement(_core.Flex, {
    justifyContent: "space-between",
    mx: "20px"
  }, /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "14px",
    color: "#A0AEC0"
  }, left), /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "14px",
    color: "#A0AEC0"
  }, right))), /*#__PURE__*/_react.default.createElement(_core.Box, {
    mx: "20px"
  }, children));
};

var _default = SingleAnnouncement;
exports.default = _default;
