// viviui/dist/components/singleAnnouncement/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const SingleAnnouncement = (_ref) => {
  let {
    content,
    title,
    time,
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["content", "title", "time", "children"]);
  return /*#__PURE__*/_react.default.createElement(_core.PseudoBox, (0, _extends2.default)({
    d: "flex",
    justifyContent: "space-between",
    py: 2,
    borderBottom: "1px solid #A0AEC0",
    width: "full",
    maxWidth: "1280px",
    mx: "auto",
    px: 2
  }, props), /*#__PURE__*/_react.default.createElement(_core.Box, {
    w: "72%"
  }, /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "12px",
    color: "#A0AEC0"
  }, time), /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "22px",
    fontWeight: "bold",
    overflow: "hidden" // width='90%'
    ,
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    wordBreak: "break-word"
  }, title), /*#__PURE__*/_react.default.createElement(_core.Text, {
    fontSize: "14px",
    color: "#718096"
  }, content)), children);
};

var _default = SingleAnnouncement;
exports.default = _default;
