// viviui/dist/components/singleAnnouncement/statusButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@chakra-ui/core");

const StatusButton = (_ref) => {
  let {
    children
  } = _ref,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children"]);
  return /*#__PURE__*/_react.default.createElement(_core.Box, (0, _extends2.default)({
    rounded: "999px",
    fontSize: "14px",
    background: "red",
    my: "auto",
    h: "30px",
    px: "15px",
    color: "white"
  }, props), /*#__PURE__*/_react.default.createElement(_core.Text, {
    alignItems: "center",
    textAlign: "center",
    lineHeight: "30px"
  }, children));
};

var _default = StatusButton;
exports.default = _default;
